### ISSUE TITLE
raml-parser fails to parse type correctly with Union Type

### ISSUE THREAD

**Issue by JonathanMontane** [raml-org/raml-js-parser-2#595]
**Version: 1.1.12**

**Issue:**
When the type is implicitly declared at the block level, if the declared type is actually a union of types, the type is replaced by `string`.

**Note:** It works perfectly fine if the type is declared inside the `type` field.

**Example:**
Using the last example from the [spec on Union Type](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#union-type):

```yaml
types:
   HasHome:
     type: object
     properties:
       homeAddress: string
   Cat:
     type: object
     properties:
       name: string
       color: string
   Dog:
     type: object
     properties:
       name: string
       fangs: string       
   HomeAnimal: [ HasHome ,  Dog | Cat ]
```

**Expected:**
```js
    {
      "HomeAnimal": {
        "name": "HomeAnimal",
        "displayName": "HomeAnimal",
        "typePropertyKind": "TYPE_EXPRESSION",
        "type": [
          "HasHome",
          "Dog | Cat"
        ],
        "__METADATA__": {
          "primitiveValuesMeta": {
            "displayName": {
              "calculated": true
            }
          }
        }
      }
    }
```
**Actual:**
```js
    {
      "HomeAnimal": {
        "name": "HomeAnimal",
        "displayName": "HomeAnimal",
        "typePropertyKind": "TYPE_EXPRESSION",
        "type": [
          "string"
        ],
        "__METADATA__": {
          "primitiveValuesMeta": {
            "displayName": {
              "calculated": true
            },
            "type": {
              "insertedAsDefault": true
            }
          }
        }
      }
    }
```

**Comment by ddenisenko** [comment]
Reproducible, thanks a lot.

**Comment by ddenisenko** [comment]
Fixed. Fix will be published in 1.1.13.
